code-server/run.sh

```diff
@@ -57,7 +63,7 @@ for extension in "$${EXTENSIONLIST[@]}"; do
     continue
   fi
   printf "🧩 Installing extension $${CODE}$extension$${RESET}...\n"
-  output=$($CODE_SERVER --install-extension "$extension")
+  output=$($CODE_SERVER "$EXTENSION_ARG" --install-extension "$extension")
```

Comment:
Not an issue in this PR, but the "output" in an error scenario will most likely be bad if we don't capture stderr as well.

```suggestion
  output=$($CODE_SERVER "$EXTENSION_ARG" --install-extension "$extension" 2>&1)
```

Reply:
@mafredri - this actually hides the error. Current experience does look right

<img width="860" alt="Screenshot 2024-04-04 at 12 38 10 AM" src="https://github.com/coder/modules/assets/5442469/3f3f6daf-d4f2-46f0-b11b-b4116ad6446b">

Reply:
@michaelbrewer that error message is different, though. Failed installing vs failed to install. Might you have removed `$output` from the message as well?

Reply:
I was just doing negative testing to verify if I saw a failure message. 

When i add the redirection, no error is returned. So the change you suggested is not enough. 

I can look at alternatives if needed.

Reply:
I am referring to the suggested change:

`output=$($CODE_SERVER "$EXTENSION_ARG" --install-extension "$extension" 2>&1)`

Which when there is an error, nothing is printed out.

Leaving the code as is works for me, i see errors when there is one, and otherwise it works as before.

Reply:
@michaelbrewer alright, right you are. Seems this is actually a bug in `code-server`. It doesn't exit with a non-zero exit code when this happens. As such the `if` below won't ever work either.

This needs to be fixed in code-server but for now, lets leave out the 2>&1 and add a comment here explaining it doesn't work as intended due to bug.

Reply:
https://github.com/coder/code-server/issues/6367

Reply:
This should actually be fixed since v4.20.1 of code-server. Which version were you testing @michaelbrewer ?

Reply:
code-server --version returns: `4.22.1 760d1318e98945d05133c6121f99541ca7a39bf8 with Code 1.87.2`

```bash
CODE_SERVER=/home/local/.code-server/bin/code-server 
$CODE_SERVER --version
4.22.1 760d1318e98945d05133c6121f99541ca7a39bf8 with Code 1.87.2

EXTENSION_ARG=--extensions-dir=/home/local/.code-server/extenions/
extension=not-found

output=$($CODE_SERVER "$EXTENSION_ARG" --install-extension "$extension")
Extension 'not-found' not found.
Make sure you use the full extension ID, including the publisher, e.g.: ms-dotnettools.csharp
Failed Installing Extensions: not-found

echo $?
0
```

Reply:
@mafredri  - upstream M$ code does exit 1, but when i run the latest code-server it always exits 0.

Reply:
@michaelbrewer thanks for testing and reporting. Nothing we can do/improve here for the time being. (Short of detecting the text "Failed Installing Extensions", which is far from ideal.)

Reply:
Now the question is this a bug or feature. 😃   

From the end-users point of view. You get notified the extension failed to install. But the rest is fine. And you can open your workspace.